src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -28,11 +28,18 @@ public static IConfigurationBuilder AddUserSecrets<T>(this IConfigurationBuilder
             => configuration.AddUserSecrets(typeof(T).GetTypeInfo().Assembly);
 
         /// <summary>
+        /// <para>
+        /// This method is obsolete and will be removed in a future version.
+        /// The recommended alternative is <see cref="AddUserSecrets(IConfigurationBuilder, string)"/>. 
+        /// </para>
+        /// <para>
         /// Adds the user secrets configuration source. Searches the assembly from <see cref="Assembly.GetEntryAssembly"/>
         /// for an instance of <see cref="UserSecretsIdAttribute"/>.
+        /// </para>
         /// </summary>
         /// <param name="configuration"></param>
         /// <returns></returns>
+        [Obsolete("This method is obsolete and will be removed in a future version. The recommended alternative is Microsoft.Extensions.Configuration.ConfigurationExtensions.AddUserSecrets(this IConfigurationBuilder configuration, string userSecretsId).")]
```

Comment:
I thought we had agreed full signature in obsolete message, calling syntax in error message, but I'd prefer the calling syntax anyways will update.

@divega the part of the conversation that you missed is that we were talking about #575 . We figured it is better for users to upgrade to the `.AddUserSecrets(string)` overload since there is a chance we will remove the `.AddUserSecrets<T>()` overload in 2.0. You make a good point though. Pain now or possible pain later?